src/rest_api.py

```diff
@@ -60,34 +73,61 @@ def recommender():
             persist = request.args.get('persist', 'true') == 'true'
             r = RecommendationTask().execute(input_json, persist=persist,
                                              check_license=check_license)
-            status = 200
         except Exception as e:
             r = {
                 'recommendation': 'unexpected error',
                 'external_request_id': input_json.get('external_request_id'),
                 'message': '%s' % e
             }
-            status = 500
+            metrics_payload['status_code'] = 500
 
-    return flask.jsonify(r), status
+    try:
+        metrics_payload['value'] = get_time_delta(audit_data=r['result']['_audit'])
+        push_data(metrics_payload)
+    except KeyError:
+        pass
+
+    return flask.jsonify(r), metrics_payload['status_code']
 
 
 @app.route('/api/v1/stack_aggregator', methods=['POST'])
 def stack_aggregator():
     """Handle POST requests that are sent to /api/v1/stack_aggregator REST API endpoint."""
     s = {'stack_aggregator': 'failure', 'external_request_id': None}
     input_json = request.get_json()
-    current_app.logger.debug('stack_aggregator/ request with payload: {p}'.format(p=input_json))
-    if input_json and 'external_request_id' in input_json and input_json['external_request_id']:
+    metrics_payload = {
+        'pid': os.getpid(),
+        'hostname': os.environ.get("HOSTNAME"),
+        'endpoint': request.endpoint,
+        'request_method': request.method,
+        'status_code': 200
+    }
+
+    if input_json and 'external_request_id' in input_json \
+            and input_json['external_request_id']:
+
         try:
             persist = request.args.get('persist', 'true') == 'true'
             s = StackAggregator().execute(input_json, persist=persist)
+            if s is not None:
```

Comment:
YOu can add comments here that it pushes the entire stack-analyses resposne time to metrics collector. Similarly add comments wherever necessary for better maintainability and readability